consensus/clique/src/main/java/org/hyperledger/besu/consensus/clique/blockcreation/CliqueMiningCoordinator.java

```diff
@@ -33,6 +41,30 @@ public CliqueMiningCoordinator(
     this.miningTracker = miningTracker;
   }
 
+  @Override
+  protected void inSyncChanged(final boolean inSync) {
+    synchronized (this) {
+      final VoteTally validatorProvider =
+          miningTracker
+              .getProtocolContext()
+              .getConsensusState(CliqueContext.class)
+              .getVoteTallyCache()
+              .getVoteTallyAfterBlock(blockchain.getChainHeadHeader());
+      final boolean isValidator =
+          getCoinbase()
+              .filter(coinbase -> validatorProvider.getValidators().contains(coinbase))
+              .isPresent();
```

Comment:
rather than exposing the ProtocolContext, we can add a function "isValidator" to the CliqueMiningTracker?

Reply:
using getCoinbase() is kinda dangerous as it does not _necessarily_ relate to the local node's ID.

Reply:
Changed. I am now using the node's ID; Good catch